ruby/red-arrow/test/test-record-batch-raw-records.rb

```diff
@@ -0,0 +1,367 @@
+        nil
+      ]
+      epoch_date = Date.new(1970, 1, 1)
+      @date32_array = Arrow::Date32Array.new(
+        @date32_values.map {|dt| dt && (dt - epoch_date).to_i }
+      )
+
+      @date64_values = [
+        nil,
+        DateTime.new(1993,  2, 23, 15, 0, 0),
+        DateTime.new(1996, 12, 24, 15, 0, 0),
+        DateTime.new(2013,  2, 23, 15, 0, 0),
+        DateTime.new(2020, 12, 24, 15, 0, 0)
+      ]
+      @date64_array = Arrow::Date64Array.new(
+        @date64_values.map {|dt| dt && dt.to_time.gmtime.to_i * 1000 }
+      )
+
+      jst = '+09:00'
+      @timestamp_values = [
+        Time.new(1993,  2, 24, 0, 0, 0, jst).gmtime,
+        Time.new(1996, 12, 25, 0, 0, 0, jst).gmtime,
+        nil,
+        Time.new(2013,  2, 24, 0, 0, 0, jst).gmtime,
+        Time.new(2020, 12, 25, 0, 0, 0, jst).gmtime
+      ]
+      @timestamp_sec_array = Arrow::TimestampArray.new(
+        :second,
+        @timestamp_values.map {|ts| ts && ts.to_i }
+      )
+      @timestamp_msec_array = Arrow::TimestampArray.new(
+        :milli,
+        @timestamp_values.map {|ts| ts && (ts.to_r * 1_000).to_i }
+      )
+      @timestamp_usec_array = Arrow::TimestampArray.new(
+        :micro,
+        @timestamp_values.map {|ts| ts && (ts.to_r * 1_000_000).to_i }
+      )
+
+      @dictionary = Arrow::StringArray.new(['foo', 'bar', 'baz'])
+      @dictionary_indices = [0, 1, 2, 1, 0]
+      @dictionary_array = Arrow::DictionaryArray.new(
+        Arrow::DictionaryDataType.new(:int8, @dictionary, true),
+        Arrow::Int8Array.new(@dictionary_indices)
+      )
+
+      @schema = Arrow::Schema.new(
+        string: :string,
+        uint32: :uint32,
+        double: :double,
+        decimal128: @decimal128_array.value_data_type,
+        date32: :date32,
+        date64: :date64,
+        timestamp_sec: @timestamp_sec_array.value_data_type,
+        timestamp_msec: @timestamp_msec_array.value_data_type,
+        timestamp_usec: @timestamp_usec_array.value_data_type,
+        dict: @dictionary_array.value_data_type
+      )
+
+      @record_batch = Arrow::RecordBatch.new(
+        @schema, @string_values.length,
+        [
+          @string_array,
+          @uint32_array,
+          @double_array,
+          @decimal128_array,
+          @date32_array,
+          @date64_array,
+          @timestamp_sec_array,
+          @timestamp_msec_array,
+          @timestamp_usec_array,
+          @dictionary_array
+        ]
+      )
+
+      @expected_columnar_result = [
+        @string_values,
+        @uint32_values,
+        @double_values,
+        @decimal128_values.map {|d| d && BigDecimal(d) },
+        @date32_values,
+        @date64_values,
+        @timestamp_values,
+        @timestamp_values,
+        @timestamp_values,
+        @dictionary_indices
+      ]
+    end
+
+    test("default") do
+      raw_records = @record_batch.raw_records
+      assert_equal(@expected_columnar_result.transpose, raw_records)
+    end
```

Comment:
One large mixed test data and one assertion makes test difficult to maintain.

Could you split test for each type like the followings?

```ruby
  test("StringArray") do
    raw_records = [
      ["apple"],
      ["orange"],
      ["watermelon"],
      [nil],
      ["タコ"],
    ]
    record_batch = Arrow::RecordBatch.new({string: :string},
                                          raw_records)
    assert_equal(raw_records, record_batch.raw_records)
  end

  test("UInt8Array") do
    raw_records = [
      [1],
      [2],
      [4],
      [8],
      [16],
    ]
    record_batch = Arrow::RecordBatch.new({uint8: :uint8},
                                          raw_records)
    assert_equal(raw_records, record_batch.raw_records)
  end
```

Reply:
We should have a test for a record batch that contains multiple columns with different types.

Reply:
OK.
We can keep a test for multiple columns with different types but we don't need to use many columns. I think that 3 columns will be enough.
Tests for covering each type and tests for multiple columns can be isolated. Small tests are easy to maintain than a large test.

Reply:
I want to make use at least 4 columns because CRuby's Array embeds at most 3 elements.

Reply:
Then we should have both 3 columns case and 4 columns case.

Reply:
OK, I will make both cases and drop the current large case.